src/compiler/scala/tools/nsc/backend/jvm/CodeGen.scala

```diff
@@ -44,6 +45,7 @@ abstract class CodeGen[G <: Global](val global: G) extends PerRunInit {
     }
 
     genClassDefs(unit.body)
+    res.toList
```

Comment:
Naive question: could it be possible to avoid this conversion by returning `Iterator` and only converting to `List` or `Iterable` when optimizations are enabled and the code path hit `postProcessAndSendToDisk`? It seems that the happy path we're optimizing for here does only need one traversal over this data structure, in `postProcessor.generatedClasses ++= generated`.

Reply:
`ListBuffer.toList` is [almost a no-op](https://github.com/scala/scala/blob/v2.12.3/src/library/scala/collection/mutable/ListBuffer.scala#L313-L316).

Reply:
Cool, didn't know!